### PR TITLE
Image Classification Integrations Testing Implementation

### PR DESCRIPTION
**Pull request by bfineran**
integrations testing implementations for image_classification

Includes testing for train and export_onnx and sample configs for train+export and export only for resnet50-imagenette pruned as examples

### CODE REVIEW

**Comment by bfineran** [comment]
note: some of the changes to `export.py` will be made redundant after rebasing to main

**Comment by bfineran** [comment]
rebased to latest

File: tests/integrations/image_classification/test_image_classification.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                    continue  # cannot cast metric to float
+                metrics[metric] = val
+
+        assert train_args.test_args["target_name"] in metrics
+
+        metric_val = metrics[train_args.test_args["target_name"]]
+        expected_mean = train_args.test_args["target_mean"]
+        expected_std = train_args.test_args["target_std"]
+        assert (
+            expected_mean - expected_std <= metric_val <= (expected_mean + expected_std)
```
Comment (KSGulin) [inline tests/integrations/image_classification/test_image_classification.py:148]: nit
```suggestion
            (expected_mean - expected_std) <= metric_val <= (expected_mean + expected_std)
```

**Review by KSGulin — COMMENTED** [review]
Looks great! Approved once tests are passing

**Review by KSGulin — APPROVED** [review]

File: src/sparseml/pytorch/image_classification/export.py [hunk 2]
```diff
@@ -391,5 +413,22 @@ def export(
             )
 
 
+def _validate_dataset_num_classes(dataset, dataset_path, num_samples, num_classes):
```
Comment (dbogunowicz) [inline src/sparseml/pytorch/image_classification/export.py:416]: nit: type hints even though private method.

**Review by dbogunowicz — APPROVED** [review]

File: src/sparseml/pytorch/image_classification/export.py (hunk 2 above)
Comment (bfineran) [inline src/sparseml/pytorch/image_classification/export.py:416]:
```suggestion
def _validate_dataset_num_classes(
    dataset: str,
    dataset_path: str,
    num_samples: int,
    num_classes: int,
):
```

**Pull request merged**